java/src/main/scala/com/spotify/featran/java/JavaOps.scala

```diff
@@ -20,25 +20,26 @@ package com.spotify.featran.java
 import java.lang.{Double => JDouble}
 import java.util.function.BiFunction
 import java.util.{Collections, List => JList, Optional}
+import java.util.function.{Function => JFunction}
 
 import com.spotify.featran._
 import com.spotify.featran.tensorflow._
 import com.spotify.featran.xgboost._
 import ml.dmlc.xgboost4j.LabeledPoint
 import org.tensorflow.proto.example.Example
 
-import scala.collection.JavaConverters._
+
+import scala.jdk.CollectionConverters._
+import scala.compat.java8.OptionConverters._
+import scala.compat.java8.FunctionConverters._
 import scala.reflect.ClassTag
 
 private object JavaOps {
-  def requiredFn[I, O](f: SerializableFunction[I, O]): I => O =
-    (input: I) => f(input)
-
-  def optionalFn[I, O](f: SerializableFunction[I, Optional[O]]): I => Option[O] =
-    (input: I) => {
-      val o = f(input)
-      if (o.isPresent) Some(o.get()) else None
-    }
+  def requiredFn[I, O](f: JFunction[I, O]): I => O =
+    f.asScala
+
+  def optionalFn[I, O](f: JFunction[I, Optional[O]]): I => Option[O] =
+    f.asScala.andThen(_.asScala)
```

Comment:
Binary compat is broken here.
Since I change the compile order, it was not possible to reference the project's java code in scala.

This can cause issue if pple use a previous `SerializableFunction` wich is not extending `Function`

Reply:
Ah yup this is also a clear break; I would like to keep the `SerializableFunction` type here.

Reply:
Maybe we shoud just implement `SerializableFunction` as a `trait`